include/lsst/afw/geom/SkyWcs.h

```diff
@@ -0,0 +1,205 @@
+- An ast::SkyFrame with System=ICRS and standard axis order RA, Dec.
+
+If ACTUAL_PIXEL0 is present then it will be the base frame, else PIXEL0 will be the base frame.
+The SkyFrame is always the current frame.
+
+Constructors must set the following properties of the contained ast::FrameSet:
+- Include the frames listed above, with specified domains.
+    Note that AST will provide the IWC frame when reading standard WCS FITS
+    data using an ast::FitsChan if you set "IWC=1" as a property of ast::FitsChan.
+    Other constructors may have to work harder.
+- For the output ast::SkyFrame set SkyRef=CRVAL (in radians) and SkyRefIs="Ignored".
+    This makes CRVAL easy to look up, and "Ignored" prevents CRVAL from being used as an offset.
+*/
+class SkyWcs : public Transform<Point2Endpoint, SpherePointEndpoint> {
+public:
+    SkyWcs(SkyWcs const &) = delete;
+    /// Move constructor
+    SkyWcs(SkyWcs &&) = default;
+    SkyWcs &operator=(SkyWcs const &) = delete;
+    /// Move assignment operator
+    SkyWcs &operator=(SkyWcs &&) = default;
+
+    /**
+    Construct a pure tangent SkyWcs
+
+    @param[in] crpix  Center of projection on the CCD using the LSST convention:
+                        0, 0 is the lower left pixel of the parent image
+    @param[in] crval  Center of projection on the sky
+    @param[in] cdMatrix  CD matrix, where element (i-1, j-1) corresponds to FITS keyword CDi_j
+                        and i, j have range [1, 2]
+    */
+    explicit SkyWcs(Point2D const &crpix, SpherePoint const &crval, Eigen::Matrix2d const &cdMatrix);
+
+    /**
+    Construct a WCS from FITS keywords
+
+    In addition to standard FITS WCS keywords uses these LSST-specific keywords:
+    LTV1, LTV2: offset of subimage with respect to parent image
+
+    TODO implement this using readFrameSet and adding full standardization
+
+    Currently does not strip the used keywords, but that capability should probably be added
+    */
+    explicit SkyWcs(daf::base::PropertyList &metadata);
+
+    /**
+    Construct a WCS from an ast::FrameSet
+
+    This is the most general constructor; it can be used to define any celestial WCS
+    as long as it contains the @ref skywcs_frames "required frames".
+
+    @throw lsst::pex::exceptions::InvalidParameterError if `frameSet` is missing
+    any of the @ref skywcs_frames "required frames".
+    */
+    explicit SkyWcs(ast::FrameSet const & frameSet);
+
+    ~SkyWcs(){};
+
+    /**
+    Get the pixel scale at the specified pixel position
+
+    The scale is the square root of the area of the specified pixel on the sky.
+    */
+    Angle getPixelScale(Point2D const &pixel) const;
+
+    /**
+    Get CRPIX, the pixel origin, using the LSST convention
+    */
+    Point2D getPixelOrigin() const;
+
+    /**
+    Get CRVAL, the sky origin or celestial fiducial point
+    */
+    SpherePoint getSkyOrigin() const;
+
+    /**
+    Get the CD matrix at the pixel origin
```

Comment:
Perhaps you can clarify what sort of text you would like to see? I though the CD matrix was always a local linear approximation. In any case I'll say "Get the 2x2 CD matrix the pixel origin" since then it's clear it must be linear.